classes/core/PKPEventServiceProvider.inc.php

```diff
@@ -78,6 +84,82 @@ public function boot()
             Event::subscribe($subscriber);
         }
     }
+
+    /**
+     * Determine if events and listeners should be automatically discovered.
+     *
+     * @return bool
+     */
+    public function shouldDiscoverEvents()
+    {
+        return true;
+    }
+
+    /**
+     * Get the discovered events for the application.
+     *
+     * @return array
+     */
+    protected function discoveredEvents()
+    {
+        return $this->shouldDiscoverEvents()
+            ? $this->discoverEvents()
+            : [];
+    }
+
+    /**
+     * Discover the events and listeners for the application.
+     *
+     * @return array
+     */
+    public function discoverEvents()
+    {
+
+        // Adapt classes naming convention
+        $discoverEvents = new class extends DiscoverEvents
+        {
+            /**
+             * @param SplFileInfo $file
+             * @param string $basePath base path of the application
+             * @return string listener's full class name
+             */
+            protected static function classFromFile(SplFileInfo $file, $basePath): string
+            {
+                $pathFromBase = trim(Str::replaceFirst($basePath, '', $file->getRealPath()), DIRECTORY_SEPARATOR);
+                $libPath = 'lib' . DIRECTORY_SEPARATOR . 'pkp' . DIRECTORY_SEPARATOR;
```

Comment:
I think it's safe and way more readable to just use `/` in place of `DIRECTORY_SEPARATOR.

Reply:
I'm not sure here what `SplFileInfo::getRealPath()` returns on the Windows machine. `DIRECTORY_SEPARATOR` here is for correct matching when doing replacements. According to the comment here it returns backslashes: https://www.php.net/manual/en/splfileinfo.getpathname.php